src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,336 @@
+    /**
+     * Whether the check was skipped based on time/date criteria.
+     */
+    public boolean checkSkipped;
+    /**
+     * The index of the matching {@link Itinerary} as found in the OTP {@link Response} planned as part of this check.
+     */
+    public int matchingItineraryIndex = -1;
+    /**
+     * The OTP response planned to check the stored {@link Itinerary} against.
+     */
+    public Response otpResponse;
+    /**
+     * Tracks the time the notification was sent to the user.
+     */
+    public long notificationTimestamp = -1;
+
+    public CheckMonitoredTrip(MonitoredTrip trip) {
+        this.trip = trip;
+    }
+
+    /**
+     * This constructor should only be used for testing when we need to inject a mock OTP response.
+     */
+    public CheckMonitoredTrip(MonitoredTrip trip, OtpDispatcherResponse injectedOtpResponseForTesting) {
+        this.trip = trip;
+        this.injectedOtpResponseForTesting = injectedOtpResponseForTesting;
+    }
+
+    @Override
+    public void run() {
+        // Check if the trip check should be skipped (based on time, day of week, etc.)
+        checkSkipped = shouldSkipMonitoredTripCheck(trip);
+        if (checkSkipped) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = injectedOtpResponseForTesting != null
+            ? injectedOtpResponseForTesting
+            : OtpDispatcher.sendOtpPlanRequest(trip.queryParams);
+
+        if (otpDispatcherResponse.statusCode >= 400) {
+            // TODO: report bugsnag
+            LOG.error("Could not reach OTP server. status={}", otpDispatcherResponse.statusCode);
+            return;
+        }
+        otpResponse = otpDispatcherResponse.getResponse();
+        // Check monitored trip.
+        runCheckLogic();
+        // Send notifications to user. This should happen before updating the journey state so that we can check the
+        // last notification sent.
+        sendNotifications();
+        // Update journey state.
+        JourneyState journeyState = trip.retrieveJourneyState();
+        journeyState.update(this);
+    }
+
+    private void runCheckLogic() {
+        // TODO: Should null tripRequestId be fixed?
+        // TripSummary tripSummary = new TripSummary(otpDispatcherResponse.response.plan, otpDispatcherResponse.response.error, null);
+        matchingItineraryIndex = findMatchingItineraryIndex(trip, otpResponse);
+        if (matchingItineraryIndex == -1) {
+            // No matching itinerary was found.
+            enqueueNotification(TripMonitorNotification.createItineraryNotFoundNotification());
+            // TODO: Try some fancy things to construct a matching itinerary (e.g., transit index?).
```

Comment:
Another TODO here is to save to the MonitoredTrip or JourneyState that the itinerary has no longer been found. It might be nice to have this percolate to the user interface to show in the UI that the monitored trip is no longer possible. Another big TODO would be to tell _why_ the trip is no longer possible. Furthermore, in this present form monitored trips that have had an invalid itinerary will constantly have checks for whether the itinerary exists resulting in a potentially large accumulation of trip requests to OTP for trips that have impossible for a potentially long time.